### DISCUSSION TITLE
LUA Script Blocking in 5.6.23

### DISCUSSION THREAD

**Discussion by spiffo-smith** [RomanKubiak/ctrlr#491] [Q&A]
For a long time now I've been using this function for blocking scripts that are on 'luaModulatorValueChange' from running at Panel Startup:

function isPanelReady()
if panel:getBootstrapState() == false and bPanelTimedOut == 1
then return (true)
else return (false)
end
end

The bPanelTimedOut gets changed to 1 after a Timer runs that's triggered from an invisible modulator changing it's value when the panel starts. All this was extensively discussed on the previous Ctrlr website by dnaldoog and Goodweather, and in 5.3.201 this all works fine!

But after loading up my panels in 5.6.23 I found none of my LUA scripts were working off 'luaModulatorValueChange', after some digging looks like bPanelTimedOut was staying at 0 so the Timer isn't getting triggered from the invisible modulator at Panel Startup in 5.6.23 like it does in 5.3.201

Anyone else noticed this behaviour in 5.6.23 and what did they do about it?

I've just noticed that dnaldoog has posted some stuff on this very topic so I probably should have posted in there but now I can't delete this discussion!

**Comment by damiensellier** [comment]
Can you please post the link to the previous discussion/post in a comment below and click your reply as "Mark as answer"? thanks

**Reply by spiffo-smith** [reply]
This is some of the discussion on the old Ctrlr site:
https://ctrlr.org/forums/topic/standalone-images-not-loading/

I have to admit I don't really know what you mean by "Mark as answer"?

**Reply by damiensellier** [reply]
under your reply there's a button "mark as answer" like this : 

![Mark as answer](https://user-images.githubusercontent.com/55164385/192164749-9aa31f8e-e4d5-4fd9-9297-df4b0ce8c28a.png)

That's cool you found something to help your problem in the old forum. Have a nice day Spiffo ;)

**Comment by dnaldoog** [comment]
Hi Spiffo,

If you implement my class blocking function as described [here](https://github.com/RomanKubiak/ctrlr/discussions/484), you can just change your function to read:

<pre>
function isPanelReady()
return myPanel:GetBootstrapState()
end
</pre>

**Reply by spiffo-smith** [reply]
Mmmmm, yep that's a variation on what I was already doing, but do you not find that your Timer Script "unBlock" does NOT get triggered when your Panel is loaded into v5.6.23 ???

That's my question, my experience is that under v5.3.201 all the scripts on "luaModulatorValueChange" get triggered on panel startup but with v5.6.23 they do not ???

**Reply by dnaldoog** [reply]
What are the benefits of 5.6.23?

I only use 5.3.201. There are documented workarounds for 5.3.201 so it’s very stable for my needs.

> That's my question, my experience is that under v5.3.201 all the scripts on "luaModulatorValueChange" get triggered on panel startup but with v5.6.23 they do not ???

Scripts **don’t** get triggered on startup using my code in the other post in 5.3.201 for me at least.

> &#x2026; Timer Script "unBlock" does NOT get triggered when your Panel is loaded into v5.6.23 

To trigger the unblock function in 5.6.23 maybe take it out of the hidden _uiButton_ and set the timer in an init() script and make the delay 1000ms or something big?

**Reply by spiffo-smith** [reply]
I'm using 5.6.23 only to generate VST3 Restricted Instances, which I have now got to work, but that's a whole different topic!

In the end I have a 5.6.23 specific panel where I just do something very similar to running the Timer from the init() script like you say, and that seems to work fine.

It just seems a fundamental difference in the way the 2 versions work and wondered if anyone else had noticed?

**Reply by dnaldoog** [reply]
> I'm using 5.6.23 only to generate VST3 Restricted Instances, which I have now got to work, but that's a whole different topic!

Oh I see why now! 😄



I just tried creating a simple panel with a function that sets a uiSlider to value 127. The function doesn't seem to trigger at all when I load the panel like it would in 5.3.201

----
[Ctrlr Panel_5.6.23 test.panel.zip](https://github.com/RomanKubiak/ctrlr/files/9643793/Ctrlr.Panel_5.6.23.test.panel.zip)

**Reply by spiffo-smith** [reply]
> I just tried creating a simple panel with a function that sets a uiSlider to value 127. The function doesn't seem to trigger at all when I load the panel like it would in 5.3.201

Exactly, fundamentally different behaviour on Panel Load!
